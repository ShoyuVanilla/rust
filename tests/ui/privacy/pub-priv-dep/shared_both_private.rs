//@ aux-crate:priv:shared=shared.rs
//@ aux-crate:priv:reexport=reexport.rs
//@ compile-flags: -Zunstable-options

// A shared dependency, where a private dependency reexports a public dependency.
//
//         shared_both_private
//                  /\
//       (PRIVATE) /  | (PRIVATE)
//                /   |
//        reexport    |
//                \   |
//        (public) \  /
//                  \/
//                shared

#![crate_type = "lib"]
#![deny(exported_private_dependencies)]

extern crate shared;
extern crate reexport;

pub fn leaks_priv() -> shared::Shared {
//~^ ERROR type `Shared` from private dependency 'shared' in public interface
    shared::Shared
}

pub fn leaks_priv_reexport() -> reexport::Shared {
//~^ ERROR type `Shared` from private dependency 'shared' in public interface
    reexport::Shared
}
