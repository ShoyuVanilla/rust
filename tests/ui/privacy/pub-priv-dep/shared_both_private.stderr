error: type `Shared` from private dependency 'shared' in public interface
  --> $DIR/shared_both_private.rs:23:1
   |
LL | pub fn leaks_priv() -> shared::Shared {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: the lint level is defined here
  --> $DIR/shared_both_private.rs:18:9
   |
LL | #![deny(exported_private_dependencies)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type `Shared` from private dependency 'shared' in public interface
  --> $DIR/shared_both_private.rs:28:1
   |
LL | pub fn leaks_priv_reexport() -> reexport::Shared {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 2 previous errors

